dashboard/lib/tasks/lesson_plan_pdfs.rake

```diff
@@ -0,0 +1,51 @@
+namespace :lesson_plan_pdfs do
+  def get_pdf_enabled_scripts
+    Script.all.select do |script|
+      script.is_migrated && script.seeded_from.present?
+    end
+  end
+
+  def get_pdfless_lessons(script)
+    script.lessons.select(&:has_lesson_plan).select do |lesson|
+      !Services::LessonPlanPdfs.pdf_exists_for?(lesson)
+    end
+  end
+
+  desc 'Identify all content for which we expect to have a generated PDF, but don\'t.'
+  task identify_missing_pdfs: :environment do
+    any_missing_pdfs_found = false
+    get_pdf_enabled_scripts.each do |script|
+      pdfless_lessons = get_pdfless_lessons(script)
+      next if pdfless_lessons.empty?
+
+      any_missing_pdfs_found = true
+      puts "Script #{script.name.inspect} is missing PDFs for:"
+      pdfless_lessons.each do |lesson|
+        puts "  #{lesson.name.inspect} (#{lesson.key})"
+      end
+    end
+
+    puts "No missing PDFs found" unless any_missing_pdfs_found
+  end
+
+  desc 'Generate any PDFs that we would expect to have been generated automatically but for whatever reason haven\'t been.'
+  task generate_missing_pdfs: :environment do
+    Dir.mktmpdir("pdf_generation") do |dir|
+      any_pdf_generated = false
+      get_pdf_enabled_scripts.each do |script|
+        get_pdfless_lessons(script).each do |lesson|
```

Comment:
Out of curiousity, is there any downside of generating all PDFs, other than performance? This rake task seems fine for solving issues with the generation step but I'm also thinking through a scenario where we changed some styling for the pdfs and wanted to regenerate all of them. I don't think we necessarily need a rake task for this, but curious if you know of any gotchas here

Reply:
I'm pretty sure it would just be performance. Generating a single lesson PDF takes about 4 seconds, but there are enough lessons that generating everything would take about half an hour.

I'll go ahead and add a task to this for that, mostly so that I can make sure we have an adequate warning

Reply:
Makes sense -- thanks!